src/Identity/UI/src/IdentityBuilderUIExtensions.cs

```diff
@@ -36,5 +63,90 @@ public static IdentityBuilder AddDefaultUI(this IdentityBuilder builder)
 
             return builder;
         }
+
+        private static Assembly GetApplicationAssembly(IdentityBuilder builder)
+        {
+            // Whis is the same logic that MVC follows to find the application assembly.
+            var environment = builder.Services.Where(d => d.ServiceType == typeof(IWebHostEnvironment)).ToArray();
+            var applicationName = ((IWebHostEnvironment)environment.LastOrDefault()?.ImplementationInstance)
+                .ApplicationName;
+
+            if (applicationName == null)
+            {
+                return null;
+            }
+            var appAssembly = Assembly.Load(applicationName);
+            return appAssembly;
+        }
+
+        private static bool TryResolveUIFramework(Assembly assembly, out UIFramework uiFramework)
+        {
+            uiFramework = default;
+
+            var metadata = assembly?.GetCustomAttributes<UIFrameworkAttribute>()
+                .SingleOrDefault()?.UIFramework; // Bootstrap5 is the default
+            if (metadata == null)
+            {
+                return false;
+            }
+
+            // If we find the metadata there must be a valid framework here.
+            if (!Enum.TryParse(metadata, ignoreCase: true, out uiFramework))
+            {
+                var enumValues = string.Join(", ", Enum.GetNames(typeof(UIFramework)).Select(v => $"'{v}'"));
+                throw new InvalidOperationException(
+                    $"Found an invalid value for the 'IdentityUIFrameworkVersion'. Valid values are {enumValues}");
```

Comment:
Is IdentityUIFrameworkVersion the right parameter name here? I could be missing something but I don't think that's the user-facing name of this, is it?

Reply:
Yeah it is, this was what we used in the past as well, https://github.com/dotnet/aspnetcore/blob/6731f6c513ab57a10ad083c9d6a8a35936682a76/src/Identity/samples/IdentitySample.DefaultUI/IdentitySample.DefaultUI.csproj#L7

Reply:
Hmm... I saw that property, but down below, I see:
```xml
    <AssemblyAttribute Include="Microsoft.AspNetCore.Identity.UI.UIFrameworkAttribute">
      <_Parameter1>$(IdentityUIFrameworkVersion)</_Parameter1>
    </AssemblyAttribute>
```

Reply:
Yeah I asked @javiercn about that too, that extra stuff is for the project reference scenario, that's not needed for the normal package/app usage.  So most people won't be going down that path, the IdentityUIFrameworkVersion is the only mainline usage scenario

Reply:
I updated the comment to clarify that today: 
`<!-- Include an assembly attribute into the built application assembly to indicate which flavor of the UI framework to use (needed for project reference) -->`